cmd/backup-manager/app/backup/manager.go

```diff
@@ -60,10 +83,34 @@ func (bm *Manager) ProcessBackup() error {
 	if backup.Spec.BR == nil {
 		return fmt.Errorf("no br config in %s", bm)
 	}
-	return bm.performBackup(backup.DeepCopy())
+
+	bm.setOptions(backup)
+
+	var db *sql.DB
+	err = wait.PollImmediate(constants.PollInterval, constants.CheckTimeout, func() (done bool, err error) {
+		db, err = util.OpenDB(bm.GetDSN())
+		if err != nil {
+			klog.Warningf("can't connect to tidb cluster %s, err: %s", bm, err)
+			return false, nil
```

Comment:
Why return `nil` for the error case?

Reply:
https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ConditionFunc ConditionFunc returns true if the condition is satisfied, or an error if the loop should be aborted.